crates/voicevox_core/src/engine/model.rs

```diff
@@ -0,0 +1,33 @@
+#[allow(dead_code)] // TODO: remove this feature
+#[derive(Clone, Debug)]
+pub(super) struct MoraModel {
+    pub text: String,
+    pub consonant: Option<String>,
+    pub consonant_length: Option<f32>,
+    pub vowel: String,
+    pub vowel_length: f32,
+    pub pitch: f32,
+}
+
+#[allow(dead_code)] // TODO: remove this feature
+#[derive(Debug)]
+pub(super) struct AccentPhraseModel {
+    pub moras: Vec<MoraModel>,
+    pub accent: usize,
+    pub pause_mora: Option<MoraModel>,
+    pub is_interrogative: bool,
+}
+
+#[allow(dead_code)] // TODO: remove this feature
+pub(super) struct AudioQueryModel {
+    accent_phrases: Vec<AccentPhraseModel>,
+    speed_scale: f32,
+    pitch_scale: f32,
+    intonation_scale: f32,
+    volume_scale: f32,
+    pre_phoneme_length: f32,
+    post_phoneme_length: f32,
+    output_sampling_rate: u32,
+    output_stereo: bool,
+    kana: String,
+}
```

Comment:
いったんderive_newとderive_getters使ってみようという方針になっていたはずなので各structにdrive(new,Getter)をしてfieldのpubを外してstructのimmutable性を高めたほうが良いと思います
```suggestion
#[derive(Clone, Debug, new, Getter)]
pub(super) struct MoraModel {
    text: String,
    consonant: Option<String>,
    consonant_length: Option<f32>,
    vowel: String,
    vowel_length: f32,
    pitch: f32,
}

#[allow(dead_code)] // TODO: remove this feature
#[derive(Debug, new, Getter)]
pub(super) struct AccentPhraseModel {
    moras: Vec<MoraModel>,
    accent: usize,
    pause_mora: Option<MoraModel>,
    is_interrogative: bool,
}

#[allow(dead_code)] // TODO: remove this feature
#[derive(new, Getter)]
pub(super) struct AudioQueryModel {
    accent_phrases: Vec<AccentPhraseModel>,
    speed_scale: f32,
    pitch_scale: f32,
    intonation_scale: f32,
    volume_scale: f32,
    pre_phoneme_length: f32,
    post_phoneme_length: f32,
    output_sampling_rate: u32,
    output_stereo: bool,
    kana: String,
}
```

Reply:
とりあえず new と Getters を追加したのですが、ごく一部の属性は mutable に扱うことがあるため、最低限の広さで public としてみました。

https://github.com/VOICEVOX/voicevox_core/blob/42e6cfdf7ca6d8370061f56853c1d41c7a31a9db/crates/voicevox_core/src/engine/kana_parser.rs#L151-L162
https://github.com/VOICEVOX/voicevox_core/blob/42e6cfdf7ca6d8370061f56853c1d41c7a31a9db/crates/voicevox_core/src/engine/model.rs#L17-L24

Reply:
accessibilityを変えるのではなく、必要な部分にだけsetterをAccentPhraseModel に生やすように出来ますか？
機能上全てのフィールドを初期化後に変更する可能性があるのであればGettersを使わずに初めからpublicにしたほうがよいですが、今回のように部分的に変更するのであればsetterが良いと思います